test/libsolidity/syntaxTests/inlineAssembly/shadowing/function_arguments.sol

```diff
@@ -0,0 +1,11 @@
+contract C {
+    uint x;
+    function f() public pure {
+        assembly {
+            function g(f) -> x {}
```

Comment:
Technically, this wouldn't trigger the bug from the issue. But it's fine as long as such examples don't reach codegen.